libtelemetry/src/full/java/com/mapbox/android/telemetry/location/LocationCollectionClient.java

```diff
@@ -83,8 +84,9 @@ public static LocationCollectionClient install(@NonNull Context context, long de
     synchronized (lock) {
       if (locationCollectionClient == null) {
         locationCollectionClient = new LocationCollectionClient(new LocationEngineControllerImpl(applicationContext,
-          LocationEngineProvider.getBestLocationEngine(applicationContext), new SessionIdentifier(defaultInterval)),
+          LocationEngineProvider.getBestLocationEngine(applicationContext)),
           new HandlerThread("LocationSettingsChangeThread"),
+          new SessionIdentifier(defaultInterval),
           applicationContext.getSharedPreferences(MAPBOX_SHARED_PREFERENCES, Context.MODE_PRIVATE),
```

Comment:
We should use `TelemetryUtils.obtainSharedPreferences` here

Reply:
It is intentional: `TelemetryUtils` should not be public, the plan is to downgrade its visibility to package private at some point. Hence i prefer to avoid creating any dependencies on it going forward.

Reply:
Ok, but keep in mind we may have strict mode issue here.

Reply:
Are you suggesting to access it on the background thread?

Reply:
Yeah, my thought is that we could create a sharedPreference later and put all background threads in it.

Reply:
i need shared preferences at startup but i will make sure there's no impact on cold start by deferring init to background thread.